crates/compiler/load_internal/src/file.rs

```diff
@@ -4047,6 +4086,81 @@ fn parse_header<'a>(
     }
 }
 
+fn load_packages<'a>(
+    packages: &[Loc<PackageEntry<'a>>],
+    load_messages: &mut Vec<Msg<'a>>,
+    roc_cache_dir: RocCacheDir,
+    cwd: PathBuf,
+    arena: &'a Bump,
+    module_id: ModuleId,
+    module_ids: Arc<Mutex<PackageModuleIds<'a>>>,
+    ident_ids_by_module: SharedIdentIdsByModule,
+) {
+    // Load all the packages
+    for Loc { value: entry, .. } in packages.iter() {
+        let PackageEntry {
+            shorthand,
+            package_name:
+                Loc {
+                    value: package_name,
+                    ..
+                },
+            ..
+        } = entry;
+
+        let src = package_name.to_str();
+
+        // find the `package` or `platform` module on disk,
+        // downloading it into a cache dir first if necessary.
+        let root_module_path = if src.starts_with("https://") {
+            #[cfg(not(target_family = "wasm"))]
+            {
+                // If this is a HTTPS package, synchronously download it
+                // to the cache before proceeding.
+
+                // TODO we should do this async; however, with the current
+                // architecture of file.rs (which doesn't use async/await),
+                // this would be very difficult!
+                let (package_dir, opt_root_module) = cache::install_package(roc_cache_dir, src)
+                    .unwrap_or_else(|err| {
+                        todo!("TODO gracefully handle package install error {:?}", err);
+                    });
+
+                // You can optionally specify the root module using the URL fragment,
+                // e.g. #foo.roc
+                // (defaults to main.roc)
+                match opt_root_module {
+                    Some(root_module) => package_dir.join(root_module),
+                    None => package_dir.join("main.roc"),
+                }
+            }
+
+            #[cfg(target_family = "wasm")]
+            {
+                panic!("Specifying packages via URLs is curently unsupported in wasm.");
```

Comment:
Interesting! They are .roc source code, right? Why does the target matter?

Reply:
The issue is that the Rust HTTP library we're using right now doesn't support wasm. Presumably there's some way to get that to work, but I dunno if people are going to be downloading many packages in the Web repl anyway. 😄